apps/glusterfs/volume_entry_allocate.go

```diff
@@ -205,14 +206,30 @@ func (v *VolumeEntry) replaceBrickInVolume(db *bolt.DB, executor executors.Execu
 		}
 
 		// Try to allocate a brick on this device
-		newBrickEntry := newDeviceEntry.NewBrickEntry(oldBrickEntry.Info.Size,
-			float64(v.Info.Snapshot.Factor),
-			v.gidRequested, v.Info.Id)
+		err = db.Update(func(tx *bolt.Tx) error {
+			deviceEntry, _ := NewDeviceEntryFromId(tx, deviceId)
+			newBrickEntry = deviceEntry.NewBrickEntry(oldBrickEntry.Info.Size,
+				float64(v.Info.Snapshot.Factor),
+				v.gidRequested, v.Info.Id)
+			deviceEntry.Save(tx)
+			return nil
+		})
 
```

Comment:
Now you re-fetched the device entry in the transaction, and changed it,  you should re-fetch it outside? Otherwise the next operationon newDeviceEntry will operate on stale data?